pysoa/server/server.py

```diff
@@ -171,7 +171,7 @@ def handle_error(self, error, variables=None):
         except Exception:
             error_str, traceback_str = "Error formatting error", traceback.format_exc()
         # Log what happened
-        self.logger.error("Unhandled error: %s", traceback_str)
+        self.logger.error(error, exc_info=True)
```

Comment:
This isn't going to work as expected, because it's no longer within the scope of the caught exception. `sys.exc_info()` will return `None, None, None` once the exception is no longer being handled. It's fine below, but not here. Here, you'll need to formulate an exception tuple (`exception_type, exception_value, traceback`) and pass that to `exc_info` like so:

    try:
        exc_tuple = [exception_type], [exception_value], traceback.format_exc()
        ...
    except Exception as e:
        exc_tuple = type(e), e, traceback.format_exc()
        ...
    self.logger.error(error, exc_info=exc_tuple)

Reply:
I wasn't sure whether or not it would work when I first tried it, but it does. Since `handle_error` is only ever called within an exception context, `sys.exc_info()` returns the traceback as expected.

I also did a sanity check by raising an exception inside an action and adding `print sys.exc_info()` at line 173, and got an exception tuple as expected. It really does work that way.

Reply:
TIL